cmds/report.c

```diff
@@ -20,6 +21,213 @@ enum {
+	.id      = REPORT_F_TOTAL_TIME,
+	.name    = "total-time",
+	.alias   = "total",
+	.header  = "Total time",
+	.length  = 10,
+	.print   = print_total_time,
+	.list    = LIST_HEAD_INIT(field_total_time.list),
+};
+
+static struct display_field field_total_time_avg = {
+	.id      = REPORT_F_TOTAL_TIMEAVG,
+	.name    = "total-time-avg",
+	.alias   = "total-avg",
+	.header  = " Total avg",
+	.length  = 10,
+	.print   = print_total_time_avg,
+	.list    = LIST_HEAD_INIT(field_total_time_avg.list),
+};
+
+static struct display_field field_total_time_min = {
+	.id      = REPORT_F_TOTAL_TIMEMIN,
+	.name    = "total-time-min",
+	.alias   = "total-min",
+	.header  = " Total min",
+	.length  = 10,
+	.print   = print_total_time_min,
+	.list    = LIST_HEAD_INIT(field_total_time_min.list),
+};
+
+static struct display_field field_total_time_max = {
+	.id      = REPORT_F_TOTAL_TIMEMAX,
+	.name    = "total-time-max",
+	.alias   = "total-max",
+	.header  = " Total max",
+	.length  = 10,
+	.print   = print_total_time_max,
+	.list    = LIST_HEAD_INIT(field_total_time_max.list),
+};
+
+static struct display_field field_self_time = {
+	.id      = REPORT_F_SELF_TIME,
+	.name    = "self-time",
+	.alias   = "self",
+	.header  = " Self time",
+	.length  = 10,
+	.print   = print_self_time,
+	.list    = LIST_HEAD_INIT(field_self_time.list),
+};
+
+static struct display_field field_self_time_avg = {
+	.id      = REPORT_F_SELF_TIMEAVG,
+	.name    = "self-time-avg",
+	.alias   = "self-avg",
+	.header  = "  Self avg",
+	.length  = 10,
+	.print   = print_self_time_avg,
+	.list    = LIST_HEAD_INIT(field_self_time_avg.list),
+};
+
+static struct display_field field_self_time_min = {
+	.id      = REPORT_F_SELF_TIMEMIN,
+	.name    = "self-time-min",
+	.alias   = "self-min",
+	.header  = "  Self min",
+	.length  = 10,
+	.print   = print_self_time_min,
+	.list    = LIST_HEAD_INIT(field_self_time_min.list),
+};
+
+static struct display_field field_self_time_max = {
+	.id      = REPORT_F_SELF_TIMEMAX,
+	.name    = "self-time-max",
+	.alias   = "self-max",
+	.header  = "  Self max",
+	.length  = 10,
+	.print   = print_self_time_max,
+	.list    = LIST_HEAD_INIT(field_self_time_max.list),
+};
+
+static struct display_field field_call = {
+	.id      = REPORT_F_CALL,
+	.name    = "call",
+	.alias   = "call",
+	.header  = "     Calls",
+	.length  = 10,
+	.print   = print_call,
+	.list    = LIST_HEAD_INIT(field_call.list),
+};
+
+/* index of this table should be matched to display_field_id */
+static struct display_field *field_table[] = {
+	&field_total_time,
+	&field_total_time_avg,
+	&field_total_time_min,
+	&field_total_time_max,
+	&field_self_time,
+	&field_self_time_avg,
+	&field_self_time_min,
+	&field_self_time_max,
+	&field_call,
+};
+
+static void setup_default_field(struct list_head *fields, struct opts *opts)
+{
+	add_field(fields, field_table[REPORT_F_TOTAL_TIME]);
+	add_field(fields, field_table[REPORT_F_SELF_TIME]);
+	add_field(fields, field_table[REPORT_F_CALL]);
```

Comment:
You can setup different default fields based on the avg mode.

Reply:
Thank you for the review.
It'd be good to define the specification of -f option of record now.
As you can see the change of doc file, there's no avg mode in this patch currently. The only way to choose field is -f option.  
As referring your above comment, maybe you want to maintain avg mode. so could you let me know your opinion about default fields and selectable fields for each mode? 

This is one suggestion.
1. default (in case of not using --avg-self or --avg-total)
    - default field : total, self, call
    - selectable field : total, self, call, none ?
2. avg-total mode (in case of using --avg-total)
    - default field : total-avg, total-min, total-max
    - selectable field:  total-avg, total-min, total-max, call, none
3. avg-self mode (in case of using --avg-self)
    - default field: self-avg, self-min, self-max
    - selectable field:  self-avg, self-min, self-max, call, none

Any other opinions are also welcome.

Reply:
I don't think we want to limit selectable fields - simply ignore `--avg-total/self` when it's used with `--field` or show an error.

Reply:
good. 
In short, 
1. selectable fields for -f remain the same as this patch.
2. --avg-total/self works as it does in the current master.
3. ignore --avg-total/self when it's used with --field or show an error.
is that right?

Are you guys interested in a "all" field that activates all selectable fields?

Reply:
Correct, but personally I don't want to add the "all" field as it'd be too verbose and might be confusing.